src/Admitto.Infrastructure/Jobs/JobsWorker.cs

```diff
@@ -0,0 +1,350 @@
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Failed to schedule orphaned jobs cleanup");
+        }
+    }
+
+    private async Task CheckScheduledJobsAsync(CancellationToken cancellationToken)
+    {
+        using var scope = serviceProvider.CreateScope();
+        var jobContext = scope.ServiceProvider.GetRequiredService<IJobContext>();
+        var jobRunner = scope.ServiceProvider.GetRequiredService<IJobRunner>();
+
+        var now = DateTimeOffset.UtcNow;
+        var dueJobs = await jobContext.ScheduledJobs
+            .Where(sj => sj.IsEnabled && sj.NextRunTime <= now)
+            .ToListAsync(cancellationToken);
+
+        foreach (var scheduledJob in dueJobs)
+        {
+            try
+            {
+                logger.LogInformation("Starting scheduled job {JobId} of type {JobType}", 
+                    scheduledJob.Id, scheduledJob.JobType);
+
+                // Deserialize and start the job
+                var jobType = Type.GetType(scheduledJob.JobType);
+                if (jobType == null)
+                {
+                    logger.LogError("Job type {JobType} not found for scheduled job {JobId}", 
+                        scheduledJob.JobType, scheduledJob.Id);
+                    continue;
+                }
+
+                var jobInstance = System.Text.Json.JsonSerializer.Deserialize(
+                    scheduledJob.JobData, jobType, new System.Text.Json.JsonSerializerOptions
+                    {
+                        PropertyNamingPolicy = System.Text.Json.JsonNamingPolicy.CamelCase
+                    });
+
+                if (jobInstance is IJob job)
+                {
+                    await jobRunner.StartJob(job, cancellationToken);
+
+                    // Update next run time
+                    var cronSchedule = CronExpression.Parse(scheduledJob.CronExpression);
+                    var nextRunTime = cronSchedule.GetNextOccurrence(now.DateTime, TimeZoneInfo.Utc);
+                    if (nextRunTime.HasValue)
+                    {
+                        var nextRunTimeOffset = new DateTimeOffset(nextRunTime.Value, TimeSpan.Zero);
+                        scheduledJob.UpdateNextRunTime(nextRunTimeOffset);
+                        await jobContext.SaveChangesAsync(cancellationToken);
+                    }
+                    else
+                    {
+                        logger.LogWarning("Could not calculate next run time for scheduled job {JobId}", 
+                            scheduledJob.Id);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error starting scheduled job {JobId}: {Error}", 
+                    scheduledJob.Id, ex.Message);
+            }
+        }
+    }
+
+    private async Task ReloadRunningJobsAsync(CancellationToken cancellationToken)
```

Comment:
If we change the implementation to reduce polling (see prev comment), we also need to return pending jobs ids here.

Reply:
BTW this is doing nothing useful, just counting the jobs. It should populate the in-memory list of jobs